### ISSUE TITLE
streets as borderline with 2 names

### ISSUE THREAD

**Issue by henke54** [openstreetmap-carto/openstreetmap-carto#1735]
Hi, in Belgium, there are often streets which are the borderline of 2 municipalities, and each municipality has given then a 'separate name' to that particular street(fraction).
When rendered on OSM, it looks like this ;
https://smartshare.be/index.php/s/JKe8RK0IOxLmFtz

In JOSM, one has then to name that street(fraction) with those 2 names as streetname, and also with '(street)name:left' a name, and '(street)name:right' a name ... ;
https://smartshare.be/index.php/s/SbjiJh3rQDoee4r

Is it possible to render it like this ; https://smartshare.be/index.php/s/FvUTFimu4oKZXWl
each name on the side of the street where it belongs, and one has then to tag that street only with 'name:left' and 'name:right' , and not anymore those 2 names together ? 

i also asked it at OSM forum ;
http://forum.openstreetmap.org/viewtopic.php?id=31935

**Comment by math1985** [comment]
How common is this? How do other maps render this?

I can confirm this occurs (although very rarely) in the Netherlands as well. I believe I have seen it in Poland too.

**Comment by matkoniecz** [comment]
Yes, this situations happens also (very rarely) in Poland. I am not sure what is the typical handling of this situations.

**Comment by henke54** [comment]
According joost schouppe ; There are about 25.000 uses on the map according to Taginfo: http://taginfo.openstreetmap.org/search?q=name%3Aright

http://forum.openstreetmap.org/viewtopic.php?id=31935

**Comment by matkoniecz** [comment]
And according to reply I was mistaken - this type of problem is extremely rare or not happening at all in Poland.

**Comment by math1985** [comment]
[Here](http://www.openstreetmap.org/#map=17/49.92064/18.65317) is an example in Poland (not currently tagged correctly).  The houses in gmina Jastrzębie-Zdrój have housenumbers on Kłosowa, while the houses in gmina Pawłowice have house numbers on Graniczna.

**Comment by sanderd17** [comment]
We're currently working on Addresses in Flanders, and it appears to happen quite often. It's hard to notice on a survey (outskirts of municipalities are often not very well signed, and it's not easy to compare all signs at all crossings of a street with each other).

But with the data we received from the Flemish government, we're now able to notice a lot more of those subtle cases (the obvious differences were already noticeable before). Like municipalities sticking to the old spelling, and their neighbour changing to the new spelling (f.e. Honzebrouck vs Honzebroek), using different connector letters (Diksmuidesteenweg vs Diksmuidsesteenweg), adding suffix to the name (Ter Bunderen vs Ter Bunderenstraat), or other minor changes.

Every municipality I handle seems to have at least 1 or 2 cases like these. 

Also note that in some cases, this even happens inside a municipality. Like the Biezenhof quarter of my town is divided in blocks. Then these blocks are used as streetnames, but it's obvious that every street is belonging to two blocks. This is quite unusual, but the same left-right tagging and rendering would help people find out where the address is. See http://www.openstreetmap.org/way/201244382

**Comment by pnorman** [comment]
> How do other maps render this?

This is probably the real question, which we still need some info on.

I have an example local to me, but I believe all maps I've seen pick one name or the other for that street.

**Comment by sanderd17** [comment]
Google is aweful at rendering it. Here Google just picks one name, and renders it: https://www.google.be/maps/@50.9887949,3.1712904,16z?hl=en (there's no way you can deduce addresses from it). TomTom and Michelin do the same, just picking one name. 

Compare it to current OSM data: http://www.openstreetmap.org/#map=18/50.98607/3.17229 (you can see the street has two names, but you can't easily see which name belongs to which address). Bing also handles the two names, they get combined by an "&" in the search screen, while they get rendered both on the map without the "&": http://binged.it/1XvoKjK . It gives similar information as OSM (though in OSM, it's hard-coded as a single name with a dash, which isn't the case in Bing).

Our governmental mapping agency does a better job. When you zoom in enough, the names are placed side next to the road: http://www.geopunt.be/ (enable "Basiskaart, luchtfoto's en adres"->"Basiskaarten"->"Basiskaart GRB (volledige versie)" and go to 50,9875° - 3,1720° (fill it in in the search field), I don't find a way to share  a permalink).

**Comment by mboeringa** [comment]
> I don't find a way to share a permalink

http://www.geopunt.be/kaart?viewer_url=http%3A%2F%2Fmaps.geopunt.be%2Fresources%2Fapps%2FGeopunt-kaart_app%2Findex.html%3Fid%3Dff8080815139519c015163aeac3d006f

You missed the button ;)

**Comment by sanderd17** [comment]
Hmm, the button is visible on Chrome, but not on Firefox. Strange. Anyway, it's not related to this issue.

**Comment by pnorman** [comment]
Please just provide a picture with the map you're looking at, so that it's available in the future.

**Comment by mboeringa** [comment]
> Please just provide a picture with the map you're looking at, so that it's available in the future.

![belgium_name-left_right](https://cloud.githubusercontent.com/assets/7635726/11540218/c9b4f300-992b-11e5-85f4-f6c2fb800e5d.png)

**Comment by pnorman** [comment]
This is a quite rare feature and the rendering shown would add confusion with pushing labels over other objects. There's arguments either way, but I'm inclined towards not rendering.

**Comment by SomeoneElseOSM** [comment]
(since I see that this is being talked about elsewhere again) here's how I did it:
```

   if (( keyvalues["name:left"]  ~= nil ) and
        ( keyvalues["name:right"] ~= nil )) then
       keyvalues["name"] = keyvalues["name:left"] .. " / " .. keyvalues["name:right"]
    end

```

which just looks like:

https://map.atownsend.org.uk/maps/map/map.html#zoom=19&lat=53.047442&lon=-1.315495

**Comment by henke54** [comment]
In Belgium, there is lots of those streets with 2 names ->
http://osmose.openstreetmap.fr/nl/errors/?source=3727&item=5030&class=705
I used to tag them 2 name-streets with both names, separated with a
slash,(some others separate them with a ';' or with a '-' ) and with
name:left and name:right ->
https://www.openstreetmap.org/way/25707821

But since that 'rejected propose' i tag them only with the 2 names
separated with a slash and nothing else ...


2017-10-12 8:51 GMT+02:00 SomeoneElseOSM <notifications@github.com>:

**Comment by dch0ph** [comment]
@imagico Prompted by discussion in #5216, I think it's worth re-opening / re-considering the rendering of `name:left` and `name:right`.

Usage is relatively niche, about 11k uses on `highway` (the majority of usage is on boundaries), but is important in some areas of older terraced housing where the [front door is on a footway](https://www.openstreetmap.org/#map=18/54.800271/-1.674554):

<img width="283" height="238" alt="Image" src="https://github.com/user-attachments/assets/1757aaf5-aeb2-4120-84c4-6d657a29c649" />

[The `highway=residential` between the named footways has `name:left` and `name:right` and not `name`.]

**Comment by imagico** [comment]
I am not sure if it makes sense to add this independent of a more comprehensive approach to rendering multiple names.  But if a good idea is presented how to do this i don't really mind.

Note i am definitely against simply concatenating the two names.  The left/right needs to be reflected and it needs to work on different road classes.

**Comment by dch0ph** [comment]
> Note i am definitely against simply concatenating the two names. The left/right needs to be reflected and it needs to work on different road classes.

Agreed. The simplest approach would be to use `text-dy` to displace the two labels either side. We wouldn't be able to line up the labels as in the example above (this would require access to the mapnik text `Layout` functionality). We do this already for admin boundaries, but the risk is that the labels then become visually "detached" from the road, and, as noted above, are likely to be blocked by / block other features off the road. 

The only "inline" option I can think of is to use arrows e.g. ↑ Left / ↓ Right, but generating a composite string has its own issues.

EDIT:
> I am not sure if it makes sense to add this independent of a more comprehensive approach to rendering multiple names

I feel this issue is largely orthogonal. Any more general approach to name from https://github.com/openstreetmap-carto/openstreetmap-carto/issues/5216 can be applied to `name:left` and `name:right`. As you might expect, there are scattering of tags such as `official_name:left` and `name:left:LA`.
